server/routes/api/place.py

```diff
@@ -819,65 +819,81 @@ def api_ranking_chart(dcid):
             return Response(json.dumps(result),
                             200,
                             mimetype='application/json')
-
     configs = get_ranking_chart_configs()
     # Get the first stat var of each config.
     stat_vars, _ = shared_api.get_stat_vars(configs)
     # Make sure POPULATION_DCID is included in stat vars.
     if POPULATION_DCID not in stat_vars:
         stat_vars.add(POPULATION_DCID)
-    sv_data = dc.get_stat_set_within_place(parent_place_dcid, place_type,
-                                           list(stat_vars), "")
-    sv_data_values = sv_data.get("data", {})
-
-    if not sv_data or not sv_data_values:
+    points_response_best = dc.points_within(parent_place_dcid, place_type,
+                                            list(stat_vars), "", False)
+    sv_data = points_response_best.get("observationsByVariable")
+    sv_facets = points_response_best.get("facets")
+    if not points_response_best or not sv_data:
         return Response(json.dumps(result), 200, mimetype='application/json')
+    # Build a dict to map stat vars to dicts mapping place dcids to their data.
+    sv_to_data_points = {}
```

Comment:
Is this intermediate data structure necessary or all the following logic can just happen while looping through the original response?

Reply:
Per offline discussion, I replaced the intermediate data structure with a dict mapping config svs to dicts of unit and scaling. Instead of looping through the configs, I loop through the sv in the original response data and check against the map.